### PR TITLE
Labels support for Network, Endpoint Create, Join

### PR DESCRIPTION
**Pull request by aboch**
- Support end-to-end, along with current options
- Please comment on labels choice and location

Deprecated in favor of #222

### CODE REVIEW

File: drivers/bridge/labels.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
+package bridge
+
+const (
+	// BridgeName label for bridge driver
+	BridgeName = "io.docker.network.bridge.name"
```
Comment (thaJeztah) [inline drivers/bridge/labels.go]: I know other projects, such as Swarm and Docker Compose, are using `com.docker.<projectname>` as namespace for labels. For consistency, I think `com.docker` in stead of `io.docker` should be used. Possibly `libnetwork` as well, but less opinionated on that.


Should there be a constant defining the namespace as well?

File: drivers/bridge/labels.go [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+	// EnableIPTables label for bridge driver
+	EnableIPTables = "io.docker.network.bridge.enable_iptables"
+
+	// EnableIPMasquerade label for bridge driver
+	EnableIPMasquerade = "io.docker.network.bridge.enableipmasquerade"
```
Comment (thaJeztah) [inline drivers/bridge/labels.go]: This looks inconsistent with the other "enable" labels, should be `enable_ipmasquerade` (or even `enable_ip_masquerade`?).

Given that a "reverse domain" notation is used as convention, I think dashes (`'-'`) in stead of underscores may be preferred as separator between words

File: api/api_test.go [hunk 3]
```diff
@@ -240,6 +287,44 @@ func TestCreateDeleteNetwork(t *testing.T) {
 	if errRsp != &successResponse {
 		t.Fatalf("Unexepected failure: %v", errRsp)
 	}
+
+	// Create with labels
+	labels := []string{
+		netlabel.EnableIPv6, "true",
+		"io.docker.network.bridgename", "abc",
+		"io.docker.network.bridge.io.docker.network.bridge.allow_non_default_bridge", "true",
```
Comment (thaJeztah) [inline api/api_test.go]: Reading quickly over the code, but these are meant as "failing" tests?  If not; these labels don't match the names defined earlier :smile:

File: drivers/bridge/labels.go (hunk 1 above)
Comment (aboch) [inline drivers/bridge/labels.go]: Agree, these should be consistent with other docker projects.

File: drivers/bridge/labels.go (hunk 2 above)
Comment (aboch) [inline drivers/bridge/labels.go]: Yep, thanks.

File: api/api_test.go (hunk 3 above)
Comment (aboch) [inline api/api_test.go]: Thanks for the catch !

Comment (thaJeztah) [inline api/api_test.go]: But the tests didn't fail, so might be worth checking why?

**Comment by aboch** [comment]
Deprecated in favor of #222 Change in logic.

**Pull request closed without merging**